examples/inspection/plot_permutation_importance_SHAP.py

```diff
@@ -0,0 +1,336 @@
+"""
+
+# %%
+# Data
+# ----
+#
+# We will use the :ref:`california_housing_dataset` for this example. In this
+# dataset the target is the median house price (in $100,000's) for a district
+# and the 8 features consist of various information about each district. We
+# will only use a subset of the data to speed up computation.
+
+import pandas as pd
+
+from sklearn.datasets import fetch_california_housing
+from sklearn.model_selection import train_test_split
+
+cal_housing = fetch_california_housing()
+y = cal_housing.target[::10]
+X = pd.DataFrame(data=cal_housing.data[::10, :], columns=cal_housing.feature_names)
+X.head()
+
+# %%
+# Split the data into training and testing subsets.
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=7)
+
+# %%
+# Calculating feature importance
+# ------------------------------
+#
+# We will used a :class:`~sklearn.ensemble.RandomForestRegressor` and fit the
+# model with our training subset.
+
+from sklearn.ensemble import RandomForestRegressor
+
+reg = RandomForestRegressor(max_depth=10, random_state=7)
+reg.fit(X_train, y_train)
+
+# %%
+# Permutation importance
+# ^^^^^^^^^^^^^^^^^^^^^^
+#
+# First, we will calculate permutation importances on the test subset
+# using the default score metric of
+# :class:`~sklearn.ensemble.RandomForestRegressor`: :ref:`R² <r2_score>`.
+# The values of each feature will be permuted ``n_repeats=10`` times and the
+# decrease in R^2 value for each permutation is shown below with boxplots.
+
+import matplotlib.pyplot as plt
+
+from sklearn.inspection import permutation_importance
+
+perm_import = permutation_importance(
+    reg, X_test, y_test, n_repeats=10, random_state=42, n_jobs=2
+)
+sorted_idx = perm_import.importances_mean.argsort()
+
+fig, ax = plt.subplots()
+ax.boxplot(
+    perm_import.importances[sorted_idx].T,
+    vert=False,
+    labels=X_train.columns[sorted_idx],
+)
+ax.set_title("Permutation Importances")
+fig.tight_layout()
+plt.show()
+
+# %%
+# The plot shows that ``MedInc`` (median income) causes by far the biggest
+# drop in R² score whereas ``AveBedrms`` and ``Population`` seem to have
+# almost no effect.
+#
+# **Considerations**
+#
+# * Permutation importance is linked to the score metric used, thus selecting
+#   an appropriate metric for your needs is essential.
+# * Permutation importance can be calculated with either the training or
+#   testing subset and each provides different information about the model.
+#   Using the training subset shows what the model has learned to use for
+#   making predictions. Using the testing subset shows what is actually useful
+#   when making predictions on novel data. Features that the model is able to
+#   use to overfit on the training data will appear more important when using
+#   the training data than the testing data. This can be seen in
+#   :ref:`sphx_glr_auto_examples_inspection_plot_permutation_importance.py`
+#   where random features are added into the dataset.
+#
```

Comment:
Just a note for a later more detailed review. We should mention that the analysis is marginal with the permutation importance while conditional in the case of SHAP.